src/gun_http2.erl

```diff
@@ -970,7 +970,43 @@ headers(State, RealStreamRef=[StreamRef|_], ReplyTo, Method, _Host, _Port,
 			{[], CookieStore0, EvHandlerState0}
 	end.
 
-request(State0=#http2_state{socket=Socket, transport=Transport, opts=Opts,
+request(State=#http2_state{http2_machine=HTTP2Machine}, StreamRef, ReplyTo, Method, Host, Port,
+		Path, Headers, Body, InitialFlow, CookieStore, EvHandler, EvHandlerState)
+		when is_reference(StreamRef) ->
+
+	case cow_http2_machine:is_remote_concurrency_limit_reached(HTTP2Machine) of
+		true ->
+			ReplyTo ! {gun_error, self(), stream_ref(State, StreamRef),
+				too_many_streams},
```

Comment:
Instead of `too_many_streams` it should be `{stream_error, too_many_streams, 'Maximum concurrency limit has been reached.'}` or similar.

Reply:
Done.